### DISCUSSION TITLE
Installation Issue on a system with ROS 2 Humble installed

### DISCUSSION THREAD

**Discussion by swarajpeppermint** [Textualize/textual#3953] [Q&A] [answered]
Have you checked closed issues? https://github.com/Textualize/textual/issues?q=is%3Aissue+is%3Aclosed

**Description of Issue**
Tried to install textual on an Ubuntu 22.04 system with ROS 2 Humble installed. Had a bunch of conflicts during installation and the installation *failed*.

Install command: `python3 -m pip install textual`. Also tried `python3 -m pip install textual-dev` with the same results.

Output (See last line and the line before):

Defaulting to user installation because normal site-packages is not writeable
Collecting textual-dev
  Downloading textual_dev-1.3.0-py3-none-any.whl.metadata (1.6 kB)
Collecting aiohttp>=3.8.1 (from textual-dev)
  Downloading aiohttp-3.9.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (7.4 kB)
Collecting click>=8.1.2 (from textual-dev)
  Downloading click-8.1.7-py3-none-any.whl.metadata (3.0 kB)
Requirement already satisfied: msgpack>=1.0.3 in /usr/lib/python3/dist-packages (from textual-dev) (1.0.3)
Requirement already satisfied: textual>=0.36.0 in ./.local/lib/python3.10/site-packages (from textual-dev) (0.46.0)
Requirement already satisfied: typing-extensions<5.0.0,>=4.4.0 in ./.local/lib/python3.10/site-packages (from textual-dev) (4.9.0)
Requirement already satisfied: attrs>=17.3.0 in /usr/lib/python3/dist-packages (from aiohttp>=3.8.1->textual-dev) (21.2.0)
Collecting multidict<7.0,>=4.5 (from aiohttp>=3.8.1->textual-dev)
  Downloading multidict-6.0.4-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (114 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 114.5/114.5 kB 2.3 MB/s eta 0:00:00
Collecting yarl<2.0,>=1.0 (from aiohttp>=3.8.1->textual-dev)
  Downloading yarl-1.9.4-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (31 kB)
Collecting frozenlist>=1.1.1 (from aiohttp>=3.8.1->textual-dev)
  Downloading frozenlist-1.4.1-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (12 kB)
Collecting aiosignal>=1.1.2 (from aiohttp>=3.8.1->textual-dev)
  Downloading aiosignal-1.3.1-py3-none-any.whl (7.6 kB)
Requirement already satisfied: async-timeout<5.0,>=4.0 in ./.local/lib/python3.10/site-packages (from aiohttp>=3.8.1->textual-dev) (4.0.2)
Requirement already satisfied: markdown-it-py>=2.1.0 in ./.local/lib/python3.10/site-packages (from markdown-it-py[linkify,plugins]>=2.1.0->textual>=0.36.0->textual-dev) (3.0.0)
Requirement already satisfied: rich>=13.3.3 in ./.local/lib/python3.10/site-packages (from textual>=0.36.0->textual-dev) (13.7.0)
Requirement already satisfied: mdurl~=0.1 in ./.local/lib/python3.10/site-packages (from markdown-it-py>=2.1.0->markdown-it-py[linkify,plugins]>=2.1.0->textual>=0.36.0->textual-dev) (0.1.2)
Requirement already satisfied: linkify-it-py<3,>=1 in ./.local/lib/python3.10/site-packages (from markdown-it-py[linkify,plugins]>=2.1.0->textual>=0.36.0->textual-dev) (2.0.2)
Requirement already satisfied: mdit-py-plugins in ./.local/lib/python3.10/site-packages (from markdown-it-py[linkify,plugins]>=2.1.0->textual>=0.36.0->textual-dev) (0.4.0)
Requirement already satisfied: pygments<3.0.0,>=2.13.0 in ./.local/lib/python3.10/site-packages (from rich>=13.3.3->textual>=0.36.0->textual-dev) (2.14.0)
Requirement already satisfied: idna>=2.0 in /usr/lib/python3/dist-packages (from yarl<2.0,>=1.0->aiohttp>=3.8.1->textual-dev) (3.3)
Requirement already satisfied: uc-micro-py in ./.local/lib/python3.10/site-packages (from linkify-it-py<3,>=1->markdown-it-py[linkify,plugins]>=2.1.0->textual>=0.36.0->textual-dev) (1.0.2)
Downloading textual_dev-1.3.0-py3-none-any.whl (26 kB)
Downloading aiohttp-3.9.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.2 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.2/1.2 MB 3.7 MB/s eta 0:00:00
Downloading click-8.1.7-py3-none-any.whl (97 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 97.9/97.9 kB 2.9 MB/s eta 0:00:00
Downloading frozenlist-1.4.1-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (239 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 239.5/239.5 kB 2.8 MB/s eta 0:00:00
Downloading yarl-1.9.4-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (301 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 301.6/301.6 kB 2.7 MB/s eta 0:00:00
DEPRECATION: devscripts 2.22.1ubuntu1 has a non-standard version number. pip 24.0 will enforce this behaviour change. A possible replacement is to upgrade to a newer version of devscripts or contact the author to suggest that they release a version with a conforming version number. Discussion can be found at https://github.com/pypa/pip/issues/12063
DEPRECATION: flatbuffers 1.12.1-git20200711.33e2d80-dfsg1-0.6 has a non-standard version number. pip 24.0 will enforce this behaviour change. A possible replacement is to upgrade to a newer version of flatbuffers or contact the author to suggest that they release a version with a conforming version number. Discussion can be found at https://github.com/pypa/pip/issues/12063
Installing collected packages: multidict, frozenlist, click, yarl, aiosignal, aiohttp, textual-dev
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
streamlit 1.22.0 requires protobuf<4,>=3.12, but you have protobuf 4.23.1 which is incompatible.
Successfully installed aiohttp-3.9.1 aiosignal-1.3.1 click-8.1.7 frozenlist-1.4.1 multidict-6.0.4 textual-dev-1.3.0 yarl-1.9.4

**Comment by davep** [comment]
For those not familiar with your environment, do you think you could document what a "ROS 2 Humble" is and why it might be relevant here?

**Comment by TomJGooding** [comment, marked as answer]
> with ROS 2 Humble installed. Had a bunch of conflicts during installation and the installation failed.

Presumably whatever Python packages involved with ROS 2 Humble conflicts with some Textual dependencies? Have you tried installing Textual in a virtual environment?
